src/errors.rs

```diff
@@ -0,0 +1,42 @@
+use std::error::Error as StdError;
+use std::fmt;
+use std::io;
+use std::path::PathBuf;
+
+/// Contains an IO error that has a file path attached.
+///
+/// This type is never returned directly, but is instead wrapped inside yet
+/// another IO error.
+#[derive(Debug)]
+pub(crate) struct Error {
+    source: io::Error,
+    path: PathBuf,
```

Comment:
As I mentioned in my other comment, this should have some kind of context about what operation failed

I'm also not a fan of printing the source error message in the `Display` impl because a generic consumer who just iterates over the error causes will end up repeating the error message like
```
failed to get configuration
caused by: No such file or directory (os error 2) in /my/file
caused by: No such file or directory (os error 2)
```